bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/server/ICLanguageServerProvider.java

```diff
@@ -32,6 +33,16 @@ public interface ICLanguageServerProvider {
 	 */
 	public void setCommands(List<String> commands);
 
+	/**
+	 * Optional initialization options for the language server during. Will be put in the initialize jsonrpc call.
+	 *
+	 * @param rootUri
+	 * @return
+	 */
+	public default Object getInitializationOptions(URI rootUri) {
```

Comment:
But if say it is optional, why not to use `Optional` in signature?

Reply:
I assume by providing a default implementation, it is optional and must not be implemented by other classes.

Reply:
Null is a valid return type here.